stdlib/public/core/SmallString.swift

```diff
@@ -169,6 +174,18 @@ extension _SmallUTF8String {
   }
 }
 
+@inline(__always)
+@inlinable
+func _bytesToUInt(_ input: UnsafePointer<UInt8>, _ c: Int) -> UInt {
+  var r: UInt = 0
+  var shift: Int = 0
+  for idx in 0..<c {
+    r = r | (UInt(input[idx]) &<< shift)
+    shift = shift &+ 8
+  }
+  return r
+}
```

Comment:
Isn't this whole function just a (potentially-unaligned) load and bitfield-insert? Can you verify what code is generated here?

Reply:
The whole code is folded away (that's the point of the change). I added a test for that: SILOptimizer/string_literals.swift

Reply:
(following up), this code is also called for many other dynamic String allocations, so we'd want efficient code even when it can't be constant folded.

Reply:
Erik and I discussed changing the UBP representation. I couldn't think of any good reason to keep the (start, end] representation for the buffer, as long as the iterator keeps its representation.